crates/ra_ide/src/hover.rs

```diff
@@ -174,6 +174,8 @@ pub(crate) fn hover(db: &RootDatabase, position: FilePosition) -> Option<RangeIn
         .value
         .ancestors()
         .find(|n| ast::Expr::cast(n.clone()).is_some() || ast::Pat::cast(n.clone()).is_some())?;
+
+    // The following logic will not work if token is coming from a macro
```

Comment:
No, it is because the following line :
```
 let frange = FileRange { file_id: position.file_id, range: node.text_range() };
```
It use raw FileId instead of HirFile from macro, and the range must be wrong in MacroFile case.